### PR TITLE
chore: add centralized methods for replacing hardcoded strings for database identifiers

### PR DESCRIPTION
**Pull request by innerdvations**
### What does it do?

- Adds a `utils.identifiers` (I'm very open to suggestions on the name) export from `@strapi/database` that includes methods for getting the names of database identifiers based on the parts of the name involved (for example, content type 
- use those methods when transforming content type schemas to models for sending to the db

Note: This PR is not a breaking change (and should actually have zero tangible effects), but the follow-up will be!

Still TODO in follow-ups:

- [ ] use these methods everywhere throughout the database package (metadata/relations, metadata/index, etc)
  - [x] https://github.com/strapi/strapi/pull/19561
- [x] [add the shortening algorithm
](https://github.com/strapi/strapi/pull/19567)
- [ ] write an internal migration to handle gracefully renaming old identifiers to new ones and turn on the feature by setting the max length constant found in this PR

### Why is it needed?

To solve the [issue with long identifiers in the database](https://github.com/strapi/strapi/issues/13117) we need to construct all of our db identifiers in predictable ways with tokens that can be individually shortened (or not). The goal is to maintain as best as possible all useful information in the database identifiers while allowing nearly unlimited length attribute names and content type names.

Once every database identifier in Strapi is using these methods, we can implement an algorithm to put a max length on the identifier using a hashing function.

Note that the max length will be constant for all Strapi project in all databases rather than only when an error is thrown; this is intentional to maintain identical database schemas between environments. We do not want a project with dev sqlite (which allows unlimited length identifiers) to have different table names than production postgres (which defaults to a limit of 63), or even for a project running postgres to change their limit higher/lower and suddenly lose data.

Also note that the final PR (not this one, but stating it here to publicize intentions and not forget when the follow-up comes) will introduce a requirement that Postgres has a NAMEDATALEN >= 64 -- which is the default, so very few people should face issues -- and also include an internal migration to handle any identifiers that were previously working but need to be renamed due to the limit (ie, if a project was only running sqlite, all their identifiers could potentially be very long, and will have to be migrated to the short names)

### How to test it?

This is an initialization step that should have no effect whatseover on the actual creation of db schemas. So test it by having a Strapi instance full of data in v5/main, switching to this PR, starting up strapi, and seeing that there was no data loss.

If there was, it means there's a mistake in one of the naming methods not matching exactly what was there before, so please report exactly what data was lost so it can be fixed.

Added 100% test coverage as well

### Related issue(s)/PR(s)

https://github.com/strapi/strapi/issues/13117

### CODE REVIEW

**Review by christiancp100 — APPROVED** [review]
Code looks good to me, but I don't have enough knowledge about this domain

File: packages/core/core/src/utils/transform-content-types-to-models.ts [hunk 1]
```diff
@@ -1,12 +1,45 @@
...
+  identifiers.getTableName(collectionName, { suffix: COMPONENT_JOIN_TABLE_SUFFIX });
+
+export const getDzJoinTableName = (collectionName: string) =>
+  identifiers.getTableName(collectionName, { suffix: DZ_JOIN_TABLE_SUFFIX });
+
+const {
+  ID_COLUMN: id,
+  ORDER_COLUMN: orderColumnName,
+  FIELD_COLUMN: field,
+  ORDER_COLUMN: order,
```
Comment (Marc-Roig) [inline packages/core/core/src/utils/transform-content-types-to-models.ts:31]: why ORDER_COLUMN is used twice?

Comment (innerdvations) [reply packages/core/core/src/utils/transform-content-types-to-models.ts]: Good catch, fixed!

**Review by Marc-Roig — APPROVED** [review]
looks good!

The `indentifiers` name looks good to me, if I had to give alternatives: `databaseIdentifiers`, `columnIdentifiers`, but I would still pick `indentifiers`

For a second, after looking at the getstarted db, I thought component tables were also prefixed with `components_` but it looks like that is hardcoded into the get started schemas.

Great work!!

**Pull request merged**